### PR TITLE
Added TermsOfServiceDialog and associated stuff

### LINKED ISSUE #814: Add ToS agreement to the profile signup flow
need to:

- [x] add a checkbox to the personal form that sets `profile.agreed_to_terms_of_service`
- [x] add a dialog component that shows the actual ToS text
- [x] tests n stuff

pulled from #769

### PR DESCRIPTION
**Pull request by aliceriot**
#### What are the relevant tickets?

closes #814 

#### What's this PR do?

This moves agreement to the terms of service to the profile signup flow. Now, there's a check-box on the 'personal' step of the profile flow, and agreeing to the terms of service is required in order to progress beyond the personal step.

This also adds a dialog to show the actually text of the ToS, which can be opened by clicking a link shown next to the checkbox.

#### Where should the reviewer start?

Read over the changes to `PersonalForm.js` and the new `TermsOfServiceDialog.js`.

#### How should this be manually tested?

go to `/profile`.

1. uncheck the ToS checkbox
2. click 'save'
3. confirm that validation errors appear and you cannot move forward.
4. click on 'MIT MicroMasters Terms of Service', confirm that the dialog opens.
5. confirm that clicking 'accept' clears validation errors.

etc?

### CODE REVIEW

**Comment by noisecapella** [comment]
Is this ready for review?

**Comment by aliceriot** [comment]
almost!

**Comment by aliceriot** [comment]
@noisecapella was just doing a read-through before I marked it ready, now tis ready (for it is marked thusly)

File: static/js/components/TermsOfServiceDialog.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+              employment background; and (3) your participation and progress in MITx courses and your online
+              education.  Among other things, MIT may use the personally identifiable information that you
+              provide to respond to your questions; provide you the specific courses and/or services you select;
+              send you updates about courses and information, including specifically the MITx MicroMasters
+              program and information about MIT events; send you information about Site maintenance or updates;
+              and for all appropriate MIT administrative and research purposes. Except as set forth herein or
+              as specifically agreed to by you, MIT will not disclose any personally identifiable information
+              we gather from you on the Site to any third parties. By logging into this Site through your edX
+              account, you acknowledge and understand that personally identifiable information will be shared by
+              edX with MIT and may be used by MIT to facilitate your use of this Site.
```
Comment (pdpinch) [inline static/js/components/TermsOfServiceDialog.js]: Additional sentence here:

> In addition, please note that your education records are protected by the Family Educational Rights and Privacy Act ("FERPA") to the extent FERPA applies.

File: static/js/components/TermsOfServiceDialog.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+            </li>
+            <li>
```
Comment (pdpinch) [inline static/js/components/TermsOfServiceDialog.js]: And another bullet here (i.e. between the 2nd and 3rd):

>  Comments or other information posted by you to our forums, boards or other areas of the Site designed for public communications or communications among class members may be viewed and downloaded by others and are available to MIT faculty and staff for research and administrative purposes.  For this reason, we encourage you to use discretion when deciding to post in these forums.

**Comment by noisecapella** [comment]
Can you remove `TERMS_OF_SERVICE_URL` from `/ui/url_utils.py` and wherever it is used, and remove the terms of service Python tests?

**Comment by aliceriot** [comment]
@pdpinch thx I'll add those changes now

**Comment by aliceriot** [comment]
@pdpinch updated!

**Comment by aliceriot** [comment]
@noisecapella I deleted all the terms of service related python code I could find, did I get everything?

**Comment by pdpinch** [comment]
I thank you, and the lawyers thank you. 

> @pdpinch <https://github.com/pdpinch> updated!

**Comment by aliceriot** [comment]
oh good :D

**Comment by noisecapella** [comment]
The dialog shows a validation error though there aren't any inputs in the TOS dialog:

screenshot from 2016-08-10 16-09-31

**Comment by noisecapella** [comment]
Yes, the Python side of things looks good

**Comment by aliceriot** [comment]
Yeah, I put a validation alert around the buttons in that dialog. Should we remove it? @roberthouse54

**Comment by roberthouse54** [comment]
Yeah, I dont think we need it here.

**Comment by aliceriot** [comment]
@noisecapella Just removed the validation errors from the dialog box.

**Comment by giocalitri** [comment]
this has conflicts with master, please rebase

**Comment by noisecapella** [comment]
:+1:

**Pull request merged**